cmake/modules/SwiftUtils.cmake

```diff
@@ -95,6 +95,32 @@ function(is_build_type_optimized build_type result_var_name)
   endif()
 endfunction()
 
+function(clang_optimize_flag_for_build_type build_type result_var_name)
```

Comment:
I don't think that we should have this function.  See the question about about why this is needed.  Also this doesn't take into account MSVC currently.  The content of this function should be equivalent to:

```
  set(${result_var_name} ${CMAKE_CXX_FLAGS_${CMAKE_BUILD_TYPE}} PARENT_SCOPE)
```

Reply:
Ok, undestood. However, I don't want to switch to `CMAKE_CXX_FLAGS_${CMAKE_BUILD_TYPE}` with this PR, as that's a change to -O3 from -O2 for some presets.

Would you prefer if I instead of introducing this function kept the original code that adds -O2 for optimized builds, and after it added another -Os (last flag wins) when CMAKEBUILD_TYPE is "MinSizeRel"?

Reply:
Yeah, keeping the original would be preferable.

Reply:
https://github.com/apple/swift/pull/33388